webapp/mixins/persistentLinks.js

```diff
@@ -0,0 +1,32 @@
+export default function(options = {}) {
+  const { queryId, querySlug, path, message = 'Page not found.' } = options
+  return {
+    asyncData: async context => {
+      const {
+        params: { id, slug },
+        redirect,
+        error,
+        app: { apolloProvider }
+      } = context
+      const idOrSlug = id || slug
+
+      const variables = { idOrSlug }
+      const client = apolloProvider.defaultClient
+
+      let response
+      let resource
+      response = await client.query({ query: queryId, variables })
+      resource = response.data[Object.keys(response.data)[0]][0]
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-03-26T23:08:36Z" title="Wednesday, March 27th 2019, 12:08:36 am +01:00">Mar 27, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,32 @@
+export default function(options = {}) {
+  const { queryId, querySlug, path, message = 'Page not found.' } = options
+  return {
+    asyncData: async context => {
+      const {
+        params: { id, slug },
+        redirect,
+        error,
+        app: { apolloProvider }
+      } = context
+      const idOrSlug = id || slug
+
+      const variables = { idOrSlug }
+      const client = apolloProvider.defaultClient
+
+      let response
+      let thing
+      response = await client.query({ query: queryId, variables })
+      thing = response.data[Object.keys(response.data)[0]][0]
```


I know naming is difficult, but `thing` is not at all clear as to what it is supposed to represent

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-26T23:13:04Z" title="Wednesday, March 27th 2019, 12:13:04 am +01:00">Mar 27, 2019</time>_

---


resource?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-26T23:13:41Z" title="Wednesday, March 27th 2019, 12:13:41 am +01:00">Mar 27, 2019</time>_

---


at the moment it can be a `post` or a `user`